pyccel/stdlib/pyc_math/pyc_math.c

```diff
@@ -0,0 +1,43 @@
+#include "pyc_math.h"
+#include <math.h>
+
+/*---------------------------------------------------------------------------*/
+long        pyc_factorial(long n)
+{
+    long    res = 1;
+
+    /* ValueError: factorial() not defined for negative values */
+    if (n < 0)
+        return 0;
+    for (int i = 2; i <= n; i++)
+        res *= i;
+    return (res);
+}
+/*---------------------------------------------------------------------------*/
+long        pyc_gcd (long a, long b)
+{
+    while (b) {
+        a %= b;
+        /* swap a and b*/
+        a = a ^ b;
+        b = b ^ a;
+        a = a ^ b;
+    }
```

Comment:
I wonder if this is the fastest implementation of this function. How did you choose the implementation? Do you have a reference?

Reply:
hh, I use just a straight-forward implementation for testing purpose, I forget to mention that in the desc.
we may need to add vectorization and other optimizations to get efficient code.

Reply:
I'm not sure we need (or can) add vectorization here. But there might be a more efficient algorithm. We should probably look into this before merging otherwise we will forget to go back and do it